internal/status/status.go

```diff
@@ -22,13 +22,40 @@ var immediateGSInTerminalStatusLock = sync.Mutex{}
 // If an error occurs reporting the status, it will be logged and returned.
 //
 // This function is used by default for reporting status to the local file system unless a different method is specified.
-func ReportStatusToLocalFile(ctx *log.Context, hEnv types.HandlerEnvironment, metadata types.RCMetadata, statusType types.StatusType, c types.Cmd, msg string) error {
+func ReportStatusToLocalFile(ctx *log.Context, hEnv types.HandlerEnvironment, metadata types.RCMetadata, statusType types.StatusType, c types.Cmd, msg string, exitcode ...int) error {
 	if !c.ShouldReportStatus {
 		ctx.Log("status", "not reported for operation (by design)")
 		return nil
 	}
 
 	rootStatusJson, err := getRootStatusJson(ctx, statusType, c, msg, true, metadata.ExtName)
+	if c.Functions.Pre != nil {
+		var errorcode = constants.TranslateExitCodeToErrorClarification(exitcode[0])
+		rootStatusJson, err = getRootStatusJsonWithErrorClarification(ctx, statusType, c, msg, true, metadata.ExtName, errorcode)
+	}
+
+	if err != nil {
+		return errors.Wrap(err, "failed to get json for status report")
+	}
+
+	ctx.Log("message", "reporting status by writing status file locally")
```

Comment:
Please check whether we already have a trace stating which status we're writing. If not, then this trace should provide the status and error clarification too.